packages/types/Result.ts

```diff
@@ -38,20 +39,77 @@ export const mapHTTPError = <TError extends string>(
   return createError(mapper[err.error], err.message);
 };
 
+class ResultError<T extends ErrorResult<any>> extends Error {
+  errorResult: T;
+
+  constructor(public result: T) {
+    super(result.message);
+    this.errorResult = result;
+  }
+}
```

Comment:
Sorry if this is a dumb question but what benefit we get from implementing this as classes? Have you found limitations on the "just objects" implementation? Again not saying this is wrong, just asking for understanding.

Reply:
This class is specifically for unwrapping, so we preserve the full ErrorResult object in the thrown exception. 

Before this update the only ErrorResult data was the message string so a simple `Error` type sufficed. This type is so we can attach the full result object, including any extra data from custom ErrorResult types in the unwrapped exception

The ErrorResult type is still a plain object, as the good lord intended :-P

Reply:
>The ErrorResult type is still a plain object, as the good lord intended :-P

Amen! :pray: :laughing: